src/pyspiffe/bundle/jwt_bundle/jwt_bundle.py

```diff
@@ -51,7 +65,10 @@ def find_jwt_authority(self, key_id: str) -> _PUBLIC_KEY_TYPES:
         if not key_id:
             raise ValueError('key_id cannot be empty.')
 
-        key = self._jwt_authorities.get(key_id, None)
-        if not key:
-            raise AuthorityNotFoundError(key_id)
-        return key
+        with self.lock:
+            key = self._jwt_authorities.get(key_id)
+
+            if not key:
```

Comment:
Is it really need to keep the lock until here?

Reply:
This will be fixed as part of the changes from the other comment (raising and error vs making the return optional).
But I would like to know what do you guys think about renaming this method from `find_jwt_authority()` to `get_jwt_authority`?
For me, `get` is more accurate in this scenario where `_jwt_authority` is a dictionary.

Reply:
I like `get_`

Reply:
I like get :)... if we have put(), it makes sense to have get()... if we want get_jwt_authority, I would say let's have put_jwt_authority too. :). I prefer put/get.

Reply:
Actually `JwtBundleSet is a dictionary of JWTBundles objects, keyed by trust domain`. Why don't we just call the methods `get` & `put`?

`JwtBundleSet.get_jwt_bundle_for_trust_domain()`
vs
`JwtBundleSet.get()`

Each method has its own docstring to make its purpose clear.